modules/transport-netty4/src/test/java/org/elasticsearch/transport/netty4/NetUtilsTests.java

```diff
@@ -12,13 +12,18 @@
 import org.elasticsearch.core.IOUtils;
 import org.elasticsearch.test.ESTestCase;
 
+import java.nio.channels.SocketChannel;
+
 public class NetUtilsTests extends ESTestCase {
 
-    public void testExtendedSocketOptions() {
+    public void testExtendedSocketOptions() throws Exception {
         assumeTrue("JDK possibly not supported", Constants.JVM_NAME.contains("HotSpot") || Constants.JVM_NAME.contains("OpenJDK"));
         assumeTrue("Platform possibly not supported", IOUtils.LINUX || IOUtils.MAC_OS_X);
```

Comment:
Likewise this one - if these options are not supported on Windows then we will need to do something more involved.

Reply:
I think that one is OK though we could check that the options are not null on windows too. But IIUC, it is OK that they are not in `supportedOptions()` on windows, the original PR #59278 seemed to assume that windows does not have these options in the JDK.

Reply:
Not really familiar with how this kind of unportable thing works in Java - it looks to me that we would try and access fields like `ExtendedSocketOptions.TCP_KEEPIDLE` whether this is claimed to be supported or not, and I expect that would be reasonably fatal if this field doesn't exist.

Reply:
The `ExtendedSocketOptions` type, and therefore its fields, are present in all OpenJDK distributions, once the `jdk.net` module is resolved. The way that platform optionality is supported is through the set of socket options returned by the `NetworkChannel::supportedOptions`. For example, on Windows set of options return by `SocketChannel::supportedOptions` will not contain `TCP_KEEPIDLE`.

Reply:
I reworked the assumptions in testExtendedSocketOptions, so that various aspects of the test only run when the assumption is valid ( invalid assumptions cause the test scenario to be skipped )

Reply:
Calling this out explicitly, since I now realise that it has been implicit so far.   

With the rework to remove reflection, this code now has an explicit dependency on a JDK-specific feature, namely the extended socket options provided by the `jdk.net` module. I don't think that this is a problem, and there are few (if any) distributions of the JDK these days that are not downstream distros of OpenJDK.

Reply:
Gotcha, thanks, this makes sense now.